src/GenAPI.Desktop/Microsoft.DotNet.BuildTools.GenAPI.targets

```diff
@@ -0,0 +1,33 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <GenAPITargetPath Condition="'$(GenAPITargetPath)' == ''">$(TargetDir)$(TargetName).cs</GenAPITargetPath>
+    <_GenAPIPath>$(MSBuildThisFileDirectory)\..\tools\GenApi.exe</_GenAPIPath>
+
+    <!-- Sequence ourselves last in PrepareForRun 
+         This ensures the output has been copied to TargetPath, but still permits
+         us to participate in FileWrites. -->
+    <PrepareForRunDependsOn>
+      $(PrepareForRunDependsOn);
+      GenerateReferenceAssemblySource;
+    </PrepareForRunDependsOn>
+  </PropertyGroup>
+
+  <Target Name="GenerateReferenceAssemblySource"
```

Comment:
I specifically did not want to do that in this package.  Folks have lots of different preferences about how their projects get that metadata: properties, generated source files, checked in source files, etc.  I'd like for this package to just be responsible for the hard part: generating the source.  Folks can choose to do things like we do and check in the source (indeed a follow up change to this will be to use these targets in our own build), or they can choose to define other targets or another package to go direct to dll.  Longer term I think the direct to dll solution will be addressed better by https://github.com/dotnet/roslyn/issues/2184.